### PR TITLE
Rebalance mech lasers

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Balance "Increase the power of mech weaponry"

#### Purpose of change
Pointed out in issue #61736, the weapons of military mechs are severely underpowered compared to charge, unable to readily fight powerful and late-game enemies. As such, this PR increases the strength of the Recon and Grunt weaponry to compensate.

#### Describe the solution
Using [these](https://allcalculators.net/online-calculators/electrical-calculators/kj-to-watts-calculator/) [calculators](https://allcalculators.net/online-calculators/electrical-calculators/kj-to-watts-calculator/) to process the figures given in #61736 , I determined an estimate for damage and armor penetration for each weapon. I also increased the armor penetration due to the high energy in a small space, much like industrial laser cutters.

#### Describe alternatives you've considered
Only increasing armor penetration, or only increasing damage.

#### Testing
I spawned a mech, hacked it, and went around killing things. The changes work as intended. 

#### Additional context
None for the moment.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
/confirmed

As far as I can tell, it's caused by a failure to generate the farm tiles to the SW. Debugging I get a "context" of "mapgen_farm_lot_wire_turn_h", and searching the JSON files I end up in farm_tiles.json and farm_lots.json, where there's instructions to generate corn 90 percent of the time and wheat 10 percent.

However, rather than treating the palette parameter as a parameter, the code treats it as a palette name. Something is missing from the farm field definitions to make it resolve parameters into values.

**Comment by anoobindisguise** [comment]
are you going to make these changes apply to the standalone versions also?

**Comment by MNG-cataclysm** [comment]
> are you going to make these changes apply to the standalone versions also?

They copy from the attached mech weaponry, the changes made there will also apply to the standalone versions.

**Comment by I-am-Erk** [comment]
Would appreciate a third party review on this one, I'm not too up on gun balance stats these days

**Comment by Maleclypse** [comment]
I've requested Candlebury and Tonkatsu as reviewers either one should be able to speak to how this balances or makes sense.

**Comment by tenmillimaster** [comment]
I tried finding sources for cutting depth for pulsed lasers. The best grasp at these straws I've got is from is a brief line from the abstract of a 1992 study on IR laser surgeries
https://pubmed.ncbi.nlm.nih.gov/1405663/

> To study the immediate and long-term effects on the arterial wall, 184 craters (**1 mm diameter and 1 mm depth**) were produced perpendicular to the intimal lining in the thoracic aortas of pigs. Three types of craters were evaluated: Ho-YSGG laser-induced (lambda = 2.09 microns, 2 pulses of 500 microseconds, 0.50 joule/pulse, 50 gm force)...

This gives me an extremely rough relation between tissue* damage and energy expenditure: 1J of energy gets me a 1mm deep, 1mm wide hole in tissue*. 
(*: The tissue is pig artery, not like, flesh).

How does that relate to damage in game? I have no clue, and choosing a damage model appropriate for lasers is beyond my pay grade, but this is the kind of path I'd follow if I wanted to try and systematically decide how much damage a laser should be doing. I would decide how well porcine arterial wall approximates the average flesh of enemies in game (human, zombie, etc), and scale these numbers up, set some sort of standard ratio for volume flesh ablated: damage number in CDDA.

Without that kind of systematic approach, I really cannot comment on how much damage lasers should be doing in game. I have no real life  reference to compare to. It'll have to remain an arbitrary choice that the senior developers scale to balance the game. I'm not interested in trying to balance anything myself and am content to allow others make that creative choice in the absence of any notion of how things should be based on reality.

**Comment by Maleclypse** [comment]
https://www.govinfo.gov/content/pkg/GOVPUB-C13-2a978b9460a1df4f914933fceacb8190/pdf/GOVPUB-C13-2a978b9460a1df4f914933fceacb8190.pdf  This paper might be helpful

File: data/json/items/gun/ups.json [hunk 1]
```diff
@@ -85,7 +88,7 @@
     "color": "blue",
     "skill": "rifle",
     "range": 45,
-    "ranged_damage": { "damage_type": "heat", "amount": 30, "armor_penetration": 4 },
+    "ranged_damage": { "damage_type": "heat", "amount": 100, "armor_penetration": 30 },
```
Comment (Maleclypse) [inline data/json/items/gun/ups.json:91]:
```suggestion
    "ranged_damage": { "damage_type": "heat", "amount": 70, "armor_penetration": 30 },
```

File: data/json/items/gun/ups.json [hunk 2]
```diff
@@ -57,7 +57,7 @@
     "color": "blue",
     "skill": "rifle",
     "range": 30,
-    "ranged_damage": { "damage_type": "heat", "amount": 20, "armor_penetration": 4 },
+    "ranged_damage": { "damage_type": "heat", "amount": 40, "armor_penetration": 30 },
```
Comment (kevingranade) [inline data/json/items/gun/ups.json:60]:
```suggestion
    "ranged_damage": { "damage_type": "heat", "amount": 40 },
```

File: data/json/items/gun/ups.json [hunk 3]
```diff
@@ -69,6 +69,9 @@
   },
   {
     "id": "recon_mech_laser",
+    "//": "Presuming the beam had a diameter of 1/e2 mm coming from a 0.5 mm diameter hole, it would deliver 100% of 50 kJ.  For scale, the .50 BMG delivers <5 kJ to the target for 120 damage.",
+    "//2": "The armor penetration is hight due to the amount of energy in such a small space.  Like industrial lasers, it should cut straight through non-reflective armor.",
```
Comment (kevingranade) [inline data/json/items/gun/ups.json:73]:
```suggestion
```

File: data/json/items/gun/ups.json [hunk 4]
```diff
@@ -69,6 +69,9 @@
...
+    "//3": "Based off the .50 BMG, 1 kJ = 24 numerical damage.  Scaled down for balance reasons.",
```
Comment (kevingranade) [inline data/json/items/gun/ups.json:74]:
```suggestion
    "//"2: "Based off the .50 BMG, 1 kJ = 24 numerical damage.  Scaled down for balance reasons.",
```

File: data/json/items/gun/ups.json [hunk 5]
```diff
@@ -85,7 +88,7 @@
     "color": "blue",
     "skill": "rifle",
     "range": 45,
-    "ranged_damage": { "damage_type": "heat", "amount": 30, "armor_penetration": 4 },
+    "ranged_damage": { "damage_type": "heat", "amount": 70, "armor_penetration": 30 },
```
Comment (kevingranade) [inline data/json/items/gun/ups.json:91]:
```suggestion
    "ranged_damage": { "damage_type": "heat", "amount": 70 },
```

File: data/json/items/gun/ups.json [hunk 6]
```diff
@@ -69,6 +69,8 @@
   },
   {
     "id": "recon_mech_laser",
+    "//": "Presuming the beam had a diameter of 1/e2 mm coming from a 0.5 mm diameter hole, it would deliver 100% of 50 kJ.  For scale, the .50 BMG delivers <5 kJ to the target for 120 damage.",
+    "//"2: "Based off the .50 BMG, 1 kJ = 24 numerical damage.  Scaled down for balance reasons.",
```
Comment (Maleclypse) [inline data/json/items/gun/ups.json:73]:
```suggestion
    "//2": "Based off the .50 BMG, 1 kJ = 24 numerical damage.  Scaled down for balance reasons.",
```

**Pull request closed without merging**